source/base/Promise.ooc

```diff
@@ -65,7 +72,15 @@ Future: abstract class <T> {
 	_state: _PromiseState
 	init: func
 	wait: abstract func -> Bool
-	wait: abstract func ~default (defaultValue: T) -> T
+	wait: abstract func ~timeout (seconds: Double) -> Bool
+	wait: virtual func ~default (defaultValue: T) -> T {
+		status := this wait()
+		status ? this getResult(defaultValue) : defaultValue
+	}
+	wait: virtual func ~defaulttimeout (seconds: Double, defaultValue: T) -> T {
+		status := this wait(seconds)
+		status ? this getResult(defaultValue) : defaultValue
```

Comment:
Or on a single line:  `this wait() ? this getResult(defaultValue) : defaultValue`